### PR TITLE
Initial setup for angular service

### PR DESCRIPTION
**Pull request by Vasanth-freshworks**
-Created an NPM library for public consuming
-Created a demo application on how NPM lib works with Angular app

### CODE REVIEW

File: angular-service/demo/my-app/src/index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  <meta charset="utf-8">
+  <title>MyApp</title>
+  <base href="/">
+  <script src="https://wchat.freshchat.com/js/widget.js" async></script>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="icon" type="image/x-icon" href="favicon.ico">
+</head>
+<body>
+  <app-root></app-root>
+  <!-- <script>
```
Comment (mirage88) [inline angular-service/demo/my-app/src/index.html]: Please remove comments

File: angular-service/demo/my-app/src/app/app.component.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import { Component } from '@angular/core';
+import { Subscription } from 'rxjs';
+//import { FreshchatService } from './freshchat.service';
```
Comment (mirage88) [inline angular-service/demo/my-app/src/app/app.component.ts]: Remove

File: angular-service/demo/my-app/src/app/app.module.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+import { AppComponent } from './app.component';
+
+import { FreshchatModule } from 'angular-freshchat-widget';
```
Comment (mirage88) [inline angular-service/demo/my-app/src/app/app.module.ts]: Please add a comment above this statement informing the user to add the line in their code.

File: angular-service/lib/angular-freshchat-widget/src/freshchat.service.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  createUser = () => {
+    this.widget.fcWidget.user.create().then((resp) => {
+      this.updateUser(resp, this.user);
+    }, (err) => {
+      this.broadcastStatus({ status: 500, message: `Error creating user ${JSON.stringify(err)}` });
+      console.log(`Error creating user ${JSON.stringify(err)}`);
+    });
+  }
+
+  updateUser = (response: any, user: any) => {
```
Comment (mirage88) [inline angular-service/lib/angular-freshchat-widget/src/freshchat.service.ts]: Is this identifyUser or validateUser?

File: angular-service/demo/my-app/src/app/app.component.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+              break;
+          }
+
+        }
+      });
+  }
+  ngOnInit() {
+
+    let payload = {
+      token: "6dd0a7dd-0f28-477c-b6d4-f776a8216467",
```
Comment (mirage88) [inline angular-service/demo/my-app/src/app/app.component.ts]: Please remove prod token, and wchat.freshchat.com reference. Use placeholders.

File: angular-service/lib/angular-freshchat-widget/src/freshchat.service.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+         * Identified a mismatch between widget user and site user credentials
+         * Update restoreId for the site user
+         */
+        this.broadcastStatus({
+          status: 201,
+          message: "Identified a mismatch between widget user and site user credentials",
+          restoreId: rstId
+        });
+      }
+    }else{
```
Comment (mirage88) [inline angular-service/lib/angular-freshchat-widget/src/freshchat.service.ts]: This else block should follow the if condition **if ((externalId && externalId === extId) && (restoreId !== rstId)) {** that lies within the **if(data)**

Comment (mirage88) [reply angular-service/lib/angular-freshchat-widget/src/freshchat.service.ts]: Correct?

**Review by mirage88 — COMMENTED** [review]
Please check comments

File: angular-service/demo/my-app/src/index.html (hunk 1 above)
Comment (Vasanth-freshworks) [reply angular-service/demo/my-app/src/index.html]: Done

File: angular-service/demo/my-app/src/app/app.component.ts (hunk 2 above)
Comment (Vasanth-freshworks) [reply angular-service/demo/my-app/src/app/app.component.ts]: Done

File: angular-service/demo/my-app/src/app/app.module.ts (hunk 3 above)
Comment (Vasanth-freshworks) [reply angular-service/demo/my-app/src/app/app.module.ts]: Done

File: angular-service/demo/my-app/src/app/app.component.ts (hunk 5 above)
Comment (Vasanth-freshworks) [reply angular-service/demo/my-app/src/app/app.component.ts]: Done

File: angular-service/lib/angular-freshchat-widget/src/freshchat.service.ts (hunk 4 above)
Comment (Vasanth-freshworks) [reply angular-service/lib/angular-freshchat-widget/src/freshchat.service.ts]: Modified to verifyUser

File: angular-service/lib/angular-freshchat-widget/src/freshchat.service.ts (hunk 6 above)
Comment (Vasanth-freshworks) [reply angular-service/lib/angular-freshchat-widget/src/freshchat.service.ts]: Yep. Modified.

**Pull request closed without merging**